src/components/views/profile/index.jsx

```diff
@@ -0,0 +1,214 @@
+
+function ProfilePage() {
+  const userData = {
+    profilePicUrl: 'https://example.com/profile.jpg',
+    username: 'johndoe',
+    firstName: 'John',
+    lastName: 'Doe',
+    email: 'johndoe@example.com',
+    phoneNumber: '123-456-7890',
+    homeAddress: '123 Main St, Anytown, USA',
+    age: 30,
+  };
+
+  const [showModal, setShowModal] = useState(false);
+  const [bookSearch, setBookSearch] = useState('');
+
+  const [listedBooks, setListedBooks] = useState([
+    {
+      title: 'The Great Gatsby',
+      author: 'F. Scott Fitzgerald',
+      description: 'A classic novel set in the Jazz Age.',
+    },
+    {
+      title: 'To Kill a Mockingbird',
+      author: 'Harper Lee',
+      description: 'A story about racial injustice in the American South.',
+    },
+    {
+      title: '1984',
+      author: 'George Orwell',
+      description: 'A dystopian novel about surveillance and totalitarianism.',
+    },
+    {
+      title: 'Moby Dick',
+      author: 'Herman Melville',
+      description: "The epic tale of a captain's obsession with a white whale.",
+    },
+  ]);
+
+  const [borrowedBooks, setBorrowedBooks] = useState([
+    {
+      title: 'Pride and Prejudice',
+      author: 'Jane Austen',
+      description: 'A romantic novel set in 19th-century England.',
+    },
+    {
+      title: 'Moby Dick',
+      author: 'Herman Melville',
+      description: "The epic tale of a captain's obsession with a white whale.",
+    },
+  ]);
+
+  const [bookList, setBookList] = useState([
+    {
+      title: 'The Great Gatsby',
+      author: 'F. Scott Fitzgerald',
+      publishDate: '1925',
+      id: 1,
+    },
+    {
+      title: 'To Kill a Mockingbird',
+      author: 'Harper Lee',
+      publishDate: '1960',
+      id: 2,
+    },
+    { title: '1984', author: 'George Orwell', publishDate: '1949', id: 3 },
+    {
+      title: 'Pride and Prejudice',
+      author: 'Jane Austen',
+      publishDate: '1813',
+      id: 4,
+    },
+  ]);
+
+  const [selectedBook, setSelectedBook] = useState(null);
+  const [notes, setNotes] = useState('');
+  const [error, setError] = useState('');
+
+  const filteredBooks = bookList.filter((book) =>
+    book.title.toLowerCase().includes(bookSearch.toLowerCase())
+  );
+
+  const handleSearchChange = (e) => {
+    setBookSearch(e.target.value);
+
+    setBookList((prevBooks) =>
+      prevBooks.filter((book) =>
+        book.title.toLowerCase().includes(e.target.value.toLowerCase())
+      )
+    );
+  };
+
+  const handleButtonClick = () => {
+    if (!selectedBook) {
+      setError('You must select a book from the list.');
+      return;
+    }
+    console.log('Book listed:', selectedBook, 'with notes:', notes);
+    closeModal();
+  };
+
+  const openModal = () => setShowModal(true);
+  const closeModal = () => {
+    setShowModal(false);
+    setSelectedBook(null);
+    setNotes('');
+    setBookSearch('');
+    setError('');
+  };
+
+  return (
```

Comment:
You might want to refactor this to use the components a part of material ui, They play really nicely with each other and still support css.

Reply:
Some components to check out are Typography, Grid, Modal, and grid